osquery/tables/system/linux/startup_items.cpp

```diff
@@ -81,6 +83,113 @@ void genAutoStartScripts(const std::string& sysdir, QueryData& results) {
   }
 }
 
+int bus_iter_get_basic_and_next(DBusMessageIter* iter,
+                                int type,
+                                void* data,
+                                bool next) {
+  if (dbus_message_iter_get_arg_type(iter) != type) {
+    return -1;
+  }
+
+  dbus_message_iter_get_basic(iter, data);
+
+  if (dbus_message_iter_next(iter) != next) {
+    return -1;
+  }
+
+  return 0;
+}
+
+void genSystemDItems(QueryData& results) {
+  DBusConnection* c;
+  DBusError error;
+  DBusMessage *systemd_message = nullptr, *reply = nullptr;
+  DBusMessageIter iter, sub, sub2;
+
+  dbus_error_init(&error);
+  c = dbus_bus_get(DBUS_BUS_SYSTEM, &error);
+
+  if (c == NULL || dbus_error_is_set(&error)) {
+    VLOG(1) << "Error dbus failed: " << error.name << error.message;
```

Comment:
I am not sure what `error.name` will return, but I think you want `error.name << ": " << error.message` or something similar.